company-lsp.el

```diff
@@ -235,27 +232,31 @@ CANDIDATE is a string returned by `company-lsp--make-candidate'."
          (additional-text-edits (gethash "additionalTextEdits" item)))
     (cond
      (text-edit
-      (goto-char (line-beginning-position))
-      (delete-char (- (line-end-position) (line-beginning-position)))
-      (insert company-lsp--line-backup)
-      (lsp--apply-text-edit text-edit)
+      (setq insert-text (gethash "newText" text-edit))
+      (delete-region (- (point) (length candidate)) (point))
       (let* ((range (gethash "range" text-edit))
              (start-point (lsp--position-to-point (gethash "start" range)))
-             (new-text-length (length (gethash "newText" text-edit))))
+             (new-text-length (length insert-text)))
+        ;; set end of the range to be equal to the start of the range
+        ;; since we have already deleted the prefix.
+        (puthash "end" (gethash "start" range) range)
```

Comment:
This assumes that the text edit removes the prefix and inserts the whole completion candidate. There is no guarantee the server always sends text edit in this fashion